### PR TITLE
feat: add index_by_list command

### PR DESCRIPTION
**Pull request by tzlilamar**

### CODE REVIEW

File: intezer_analyze_cli/cli.py [hunk 1]
```diff
@@ -161,6 +161,41 @@ def analyze_by_list(path):
                    'and attach the log file in {}'.format(utilities.log_file_path))
 
 
+@main_cli.command('index_by_list', short_help='Send a text file with list of hashes, verdict, family name if malicious')
```
Comment (AlonBucks) [inline intezer_analyze_cli/cli.py:163]: great job! Just need to update version in __init__.py, add example in README.md

**Review by AlonBucks — APPROVED** [review]

**Comment by davidt99** [comment]
Add unittests

File: intezer_analyze_cli/config.py [hunk 2]
```diff
@@ -7,6 +7,7 @@ def __init__(self):
         self.api_url = 'https://analyze.intezer.com/api/'
         self.api_version = 'v2-0'
         self.analyses_url = 'https://analyze.intezer.com/#/analyses'
```
Comment (davidt99) [inline intezer_analyze_cli/config.py]: remove the hashtag

File: intezer_analyze_cli/commands.py [hunk 3]
```diff
@@ -150,6 +151,46 @@ def analyze_by_txt_file_command(path: str):
...
+                'index updated. In order to check the results go to Private Indexed Files under Analysis Reports')
+    except IOError:
+        click.echo('No read permissions for {}'.format(path))
+        click.Abort()
+
+
+def index_hash_command(sha256: str, index_as: str, family_name: Optional[str], index_exceptions):
+    try:
+        index = Index(index_as=sdk_consts.IndexType.from_str(index_as), sha256=sha256, family_name=family_name)
+        index.send(wait=True)
```
Comment (davidt99) [inline intezer_analyze_cli/commands.py]: We still need to fix this - `send(wait=False)`
After that we need to call `wait_for_completion` for each one

File: intezer_analyze_cli/cli.py [hunk 4]
```diff
@@ -151,10 +151,46 @@ def analyze_by_list(path):
...
+        raise
+    except Exception:
+        logger.exception('Unexpected error occurred')
+        click.echo('Unexpected error occurred, please contact us at support@intezer.com '
+                   'and attach the log file in {}'.format(utilities.log_file_path))
+
+
+@main_cli.command('index_by_list', short_help='Send a text file with list of hashes, verdict, family name if malicious')
+@click.argument('path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('index_as', type=click.STRING)
```
Comment (davidt99) [inline intezer_analyze_cli/cli.py]: add choice to here so it will validate the input

File: intezer_analyze_cli/cli.py [hunk 5]
```diff
@@ -151,10 +151,46 @@ def analyze_by_list(path):
...
+        if index_type == sdk_consts.IndexType.MALICIOUS and family_name is None:
+            click.echo('family_name is mandatory if the index type is malicious')
+            return
+
+        create_global_api()
+
+        commands.index_by_txt_file_command(path=path, index_as=index_as, family_name=family_name)
     except click.Abort:
         raise
+    except NotImplementedError:
```
Comment (davidt99) [inline intezer_analyze_cli/cli.py]: after the choice, this should be removed

File: intezer_analyze_cli/commands.py [hunk 6]
```diff
@@ -125,7 +125,8 @@ def analyze_directory_command(path: str,
 
 def analyze_by_txt_file_command(path: str):
     try:
-        hashes = [line.rstrip('\n') for line in open(path)]
+        with open(path, 'r') as file:
+            hashes = [line.rstrip('\n') for line in file.readlines()]
```
Comment (davidt99) [inline intezer_analyze_cli/commands.py]: let's strip

File: intezer_analyze_cli/commands.py [hunk 7]
```diff
@@ -150,6 +151,71 @@ def analyze_by_txt_file_command(path: str):
         click.Abort()
 
 
+def index_by_txt_file_command(path: str, index_as: str, family_name: str):
+    try:
+        with open(path, 'r') as file:
```
Comment (davidt99) [inline intezer_analyze_cli/commands.py]: extract if to a method, it's written twice

File: intezer_analyze_cli/commands.py [hunk 8]
```diff
@@ -150,6 +151,71 @@ def analyze_by_txt_file_command(path: str):
...
+                    index_operations.append((index_operation, sha256))
+                else:
+                    index_exceptions.append(index_exception)
+                upload_progress.update(1)
+        click.echo('Indexing sent')
+
+        echo_exceptions(index_exceptions)
+        index_exceptions = []
+        with click.progressbar(length=len(index_operations),
+                               label='Indexing files',
```
Comment (davidt99) [inline intezer_analyze_cli/commands.py]: "Waiting for indexing to finish"

File: CHANGES [hunk 9]
```diff
@@ -1,3 +1,8 @@
+1.8
+-----
+ -
```
Comment (davidt99) [inline CHANGES]: add text here

File: intezer_analyze_cli/cli.py [hunk 10]
```diff
@@ -151,8 +151,41 @@ def analyze_by_list(path):
...
+        raise
+    except Exception:
+        logger.exception('Unexpected error occurred')
+        click.echo('Unexpected error occurred, please contact us at support@intezer.com '
+                   'and attach the log file in {}'.format(utilities.log_file_path))
+
+
+@main_cli.command('index_by_list', short_help='Send a text file with list of hashes, verdict, family name if malicious')
+@click.argument('path', type=click.Path(exists=True, dir_okay=False))
+@click.option('--index-as', type=click.Choice(['malicious', 'trusted'], case_sensitive=False))
```
Comment (davidt99) [inline intezer_analyze_cli/cli.py]: i think we need it to be case sensitive

Comment (davidt99) [inline intezer_analyze_cli/cli.py]: can you do this to the other index endpoint?

**Review by davidt99 — CHANGES_REQUESTED** [review]

**Review by davidt99 — APPROVED** [review]

**Pull request merged**